lib/pages/federation_resolver.dart

```diff
@@ -4,27 +4,45 @@ import 'package:lemmy_api_client/v3.dart';
 
 import '../../l10n/l10n.dart';
 import '../../stores/accounts_store.dart';
-import 'community.dart';
 
-class FederatedCommunityPage extends HookWidget {
+class FederationResolver extends HookWidget {
   final UserData userData;
-  final String community;
+  final String query;
+  final String loadingMessage;
+  final bool Function(ResolveObjectResponse response) exists;
+  final Widget Function(BuildContext buildContext, ResolveObjectResponse object)
+      builder;
 
-  const FederatedCommunityPage(this.userData, this.community);
+  const FederationResolver({
+    required this.userData,
+    required this.query,
+    required this.loadingMessage,
+    required this.builder,
+    required this.exists,
+  });
 
   @override
   Widget build(BuildContext context) {
     final lemmyApi = LemmyApiV3(userData.instanceHost);
 
     return FutureBuilder(
-      future: lemmyApi
-          .run(ResolveObject(q: community, auth: userData.jwt.raw))
-          .then((data) {
-        Navigator.of(context).pop();
-        Navigator.of(context).push(CommunityPage.fromIdRoute(
-            userData.instanceHost, data.community!.community.id));
-      }),
+      future: lemmyApi.run(ResolveObject(q: query, auth: userData.jwt.raw)),
       builder: (context, snapshot) {
+        var message = loadingMessage;
+
+        if (snapshot.hasData) {
+          if (exists(snapshot.data!)) {
+            return Builder(
+                builder: (context) => builder(context, snapshot.data!));
+          } else {
+            message = L10n.of(context).not_found;
+          }
+        }
+
+        if (snapshot.hasError) {
+          message = 'Error: ${snapshot.error}';
```

Comment:
Out of curiosity, do we localize error messages?

Reply:
At the moment, no. At least not all of them. Even the "You are not logged in there" message isn't localized. I think maybe Lemmur was made before the l10n library could do parameters? Or they just didn't know?